### PR TITLE
Allow GitHubRunner as a project setting

### PR DESCRIPTION
**Pull request by aholstrup1**
Allow GitHubRunner as a project setting. This would enable us to compile apps on linux-based GitHub runners while running the tests on either windows GitHub runners or self-hosted runners.

### CODE REVIEW

**Comment by freddydk** [comment]
These lines at the end of ReadSettings should be removed:

```
$gitHubRunner = $settings.githubRunner.Split(',').Trim() | ConvertTo-Json -compress
Add-Content -Encoding UTF8 -Path $env:GITHUB_OUTPUT -Value "GitHubRunnerJson=$githubRunner"
Write-Host "GitHubRunnerJson=$githubRunner"

$gitHubRunnerShell = $settings.githubRunnerShell
Add-Content -Encoding UTF8 -Path $env:GITHUB_OUTPUT -Value "GitHubRunnerShell=$githubRunnerShell"
Write-Host "GitHubRunnerShell=$githubRunnerShell"
```
and these lines in the corresponding action.yaml:
```
outputs:
  GitHubRunnerJson:
    description: GitHubRunner in compressed Json format
    value: ${{ steps.readsettings.outputs.GitHubRunnerJson }}
  GitHubRunnerShell:
    description: Shell for GitHubRunner jobs
    value: ${{ steps.readsettings.outputs.GitHubRunnerShell }}
```
and these lines in the corresponding readme.md should be replaced by `none`:
```
| Name | Description |
| :-- | :-- |
| GitHubRunnerJson | GitHubRunner in compressed Json format |
| GitHubRunnerShell | Shell for GitHubRunner jobs |
```

**Review by freddydk — CHANGES_REQUESTED** [review]
Looks good - I like the change
Added a comment with requested changes (to remove the outputs from ReadSettings)

**Comment by aholstrup1** [comment]
> Looks good - I like the change Added a comment with requested changes (to remove the outputs from ReadSettings)

So.. We're still using these outputs a couple of places. Specifically, in BuildPP:
https://github.com/microsoft/AL-Go/blob/67b44892165e183e6a550a90c3d909e32604772e/Templates/Per%20Tenant%20Extension/.github/workflows/CICD.yaml#L174-L175

And CreateDevelopmentEnvironment
https://github.com/microsoft/AL-Go/blob/67b44892165e183e6a550a90c3d909e32604772e/Templates/AppSource%20App/.github/workflows/CreateOnlineDevelopmentEnvironment.yaml#L103

How do we get rid of those dependencies in an appropriate way?

File: Actions/AL-Go-TestRepoHelper.ps1 [hunk 1]
```diff
@@ -66,7 +66,6 @@ function Test-SettingsJson {
     }
     if ($type -eq 'Project') {
         # GitHubRunner should not be in a project settings file (only read from repo or workflow settings)
```
Comment (mazhelez) [inline Actions/AL-Go-TestRepoHelper.ps1:68]: Comment should be removed as well.

**Review by mazhelez — APPROVED** [review]

**Review by mazhelez — APPROVED** [review]

**Review by freddydk — APPROVED** [review]

**Pull request merged**